### ISSUE TITLE
a11y: mat-drawer-container aria-hidden elements should not be focusable. 

### ISSUE THREAD

**Issue by Anoop-D** [angular/components#16155]
#### Bug, feature request, or proposal:
Mat-drawer is composed of a focus-trap, which has the classes: cdk-visually-hidden cdk-focus-trap-anchor. 

According to WCAG 4.1.2: ARIA hidden element must not contain focusable elements.

#### What is the expected behavior?
Elements with aria-hidden should have tabindex: -1 or otherwise be unfocusable via keyboard navigation.

**Comment by crisbeto** [comment]
We have some logic that disables the focus trap [here](https://github.com/angular/components/blob/master/src/material/sidenav/drawer.ts#L400). Also note that with `tabindex` set to -1 the element is still focusable and users can reach it using the arrow keys. The way we disable the trap is by [removing the `tabindex` altogether](https://github.com/angular/components/blob/master/src/cdk/a11y/focus-trap/focus-trap.ts#L319).
